man/repart.d.xml

```diff
@@ -580,13 +582,32 @@
 
       <varlistentry>
         <term><varname>ReadOnly=</varname></term>
-
-        <listitem><para>Configures the Read-Only partition flags (bit 60) of the partition table entry. This
-        option is a friendly way to set bit 60 of the partition flags value without setting any of the other
-        bits, and may be set via <varname>Flags=</varname> too, see above.</para>
-
-        <para>If both <varname>Flags=</varname> and <varname>ReadOnly=</varname> are set the latter controls
-        the value of the flag.</para></listitem>
+        <term><varname>GrowFileSystem=</varname></term>
+
+        <listitem><para>Configures the Read-Only and Grow-File-System partition flags (bit 60 and 59) of the
+        partition table entry, as defined by the <ulink
+        url="https://systemd.io/DISCOVERABLE_PARTITIONS">Discoverable Partitions Specification</ulink>. Only
+        available for partition types supported by the specification. This option is a friendly way to set bit
+        60 and 59 of the partition flags value without setting any of the other bits, and may be set via
+        <varname>Flags=</varname> too, see above.</para>
+
+        <para>If both <varname>Flags=</varname> and
+        <varname>ReadOnly=</varname>/<varname>GrowFileSystem=</varname> are set the latter control the value
+        of the flags.</para>
```

Comment:
Shouldn't setting both ReadOnly=yes and GrowFileSystem=yes be an error instead?

Reply:
I had an error check in place for this initially, but I dropped it again. My thinking was: I think there's value if people put together images that are initially marked read-only, but where there's an explicit step where users can say "i want this to be writable now, i accept the responsibility for this". But if so, then I think it makes totally sense to make this a one-bit change, and already have other flags set (or not set) in a way that make sense then.

i.e. i want that a vendor can say "i'll mark this read-only for now, but if you change it to writable, this is what's going to happen". And by not making this flag combo an error on *write* and instead just ignoring the growfs flag if read-only is on at *read* time, we permit such a scenario.

I mean, you can construct other usecases for this too. Consider a computer forensics person wanting to make the image read-only for analysis. The way I impemented it, this can be done with a single bit change that can later be reverted back without loss of information. If we'd require this one bit change to propagate into other bits on write this would be much less sexy.

Does that make any sense?

Reply:
another probably realistic usecase: consider an installer medium that is also a "livecd". You boot it up on your installer medium and the RO bit is set. And you play around with it, but changes are not allowed, it's an installer after all. But then the installer tool dd's the whole thing onto the real HDD, i.e. doing the actual installation. And then after it's done it toggles exactly one bit on the main partition and reboots. In this scenario it#s kinda nice if the difference between "real system" and "livecd image" is really just the fact that one is read-write and the other marked read-only, and the other bits are already in place the exact way they should be after the installation.